core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/sink/wordcloud/WordCloudSink.java

```diff
@@ -0,0 +1,96 @@
+package edu.uci.ics.texera.dataflow.sink.wordcloud;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.uci.ics.texera.api.constants.ErrorMessages;
+import edu.uci.ics.texera.api.exception.DataflowException;
+import edu.uci.ics.texera.api.exception.TexeraException;
+import edu.uci.ics.texera.api.field.IntegerField;
+import edu.uci.ics.texera.api.schema.Attribute;
+import edu.uci.ics.texera.api.schema.AttributeType;
+import edu.uci.ics.texera.api.schema.Schema;
+import edu.uci.ics.texera.api.tuple.Tuple;
+import edu.uci.ics.texera.dataflow.sink.VisualizationConstants;
+import edu.uci.ics.texera.dataflow.sink.VisualizationOperator;
+
+/**
+ * WordCloudSink is a sink that can be used by the caller to generate data for wordcloud.js in frontend.
+ *
+ * @author Mingji Han
+ *
+ */
+public class WordCloudSink extends VisualizationOperator {
+    private final int MAX_FONT_SIZE = 200;
+    private final int MIN_FONT_SIZE = 50;
+    private WordCloudSinkPredicate predicate;
+    private String WORD = "word";
+    private String COUNT = "count";
+    public WordCloudSink(WordCloudSinkPredicate predicate) {
+        super(VisualizationConstants.WORD_CLOUD);
+        this.predicate = predicate;
+
+    }
+
+    @Override
+    public void open() throws TexeraException {
+        if (cursor != CLOSED) {
+            return;
+        }
+        if (inputOperator == null) {
+            throw new TexeraException(ErrorMessages.INPUT_OPERATOR_NOT_SPECIFIED);
+        }
+        inputOperator.open();
+
+        Schema schema = inputOperator.getOutputSchema();
+
+        Attribute wordColumn =  schema.getAttribute(this.WORD);
+        AttributeType nameColumnType = wordColumn.getType();
+        if (!nameColumnType.equals(AttributeType.STRING) && !nameColumnType.equals(AttributeType.TEXT)) {
+            throw new DataflowException("Type of name column should be string or text.");
+        }
+
+        Attribute countColumn =  schema.getAttribute(this.COUNT);
```

Comment:
The assumption here is that the input operator for `WordCloudSink` operator has to be `WordCount` right? (or at least contains attribute `word` and `count`). If this is the case, personally I don't think this will be a good implementation because each operator should be independent from other operators. In my opinion, `WordCloudSink` should do everything `WordCount` can do plus its own calculation of the font size( in `processTuples` function below).

Reply:
I will add two parameters `wordCoulumn` and `countColumn`  for the operator.

Reply:
Sorry I misunderstand your ideas.  
I will add word count logic in word cloud operator, so it can independent from word count operator.